src/Sylius/Bundle/CoreBundle/Validator/Constraints/MaxIntegerValidator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class MaxIntegerValidator extends ConstraintValidator
+{
+    public function __construct(private int $maxInt)
+    {
+    }
+
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        Assert::isInstanceOf($constraint, MaxInteger::class);
+
+        if ($value >= $this->maxInt) {
```

Comment:
How does it behave with string `value`?

Reply:
in that case there is type validation message at API and money validation at UI